core/src/test/java/de/danoeh/antennapod/core/feed/LocalFeedUpdaterTest.java

```diff
@@ -180,6 +184,66 @@ public void testUpdateFeed_FeedMetadata() {
         assertEquals(24, calendar.get(Calendar.SECOND));
     }
 
+    @Test
+    public void testGetImageUrl_NoImage() {
+        String defaultImageName = context.getResources().getResourceEntryName(R.raw.local_feed_default_icon);
+        {
+            // empty folder
+            DocumentFile documentFolder = mockDocumentFolder();
+            String imageUrl = LocalFeedUpdater.getImageUrl(context, documentFolder);
+            assertThat(imageUrl, endsWith(defaultImageName));
+        }
+        {
+            // no image file, with audio file
+            DocumentFile documentFolder = mockDocumentFolder(mockDocumentFile("audio.mp3", "audio/mp3"));
+            String imageUrl = LocalFeedUpdater.getImageUrl(context, documentFolder);
+            assertThat(imageUrl, endsWith(defaultImageName));
+        }
+    }
+
+    @Test
+    public void testGetImageUrl_PreferredImagesFilenames() {
+        for (String filename : LocalFeedUpdater.PREFERRED_FEED_IMAGE_FILENAMES) {
+            DocumentFile documentFolder = mockDocumentFolder(mockDocumentFile("audio.mp3", "audio/mp3"),
+                    mockDocumentFile(filename, "image/jpeg")); // image MIME type doesn't matter
+            String imageUrl = LocalFeedUpdater.getImageUrl(context, documentFolder);
+            assertThat(imageUrl, endsWith(filename));
+        }
+    }
+
+    @Test
+    public void testGetImageUrl_OtherImageFilename() {
+        {
```

Comment:
Is using anonymous blocks a common practice for tests? I think it would be a bit easier to read when just creating individual test methods for those. Then you also don't need the comment in front of each block because you can just name the tests accordingly.

Reply:
As the test cases are very similar I put them into a combined test method. The anonymous blocks were just to keep the variable lifetimes short to avoid renaming mistakes.

As suggested, I split the test methods with anonymous blocks into separate test methods now.